src/hardware/serialport/softmodem.cpp

```diff
@@ -23,12 +23,84 @@
 #include <stdlib.h>
 #include <string.h>
 #include <utility>
+#include <fstream>
+#include <sstream>
 
 #include "support.h"
 #include "serialport.h"
 #include "softmodem.h"
 #include "misc_util.h"
 
+static const char phoneValidChars[] = "01234567890*=,;#+>";
+
+class CPhonebookEntry {
+public:
+	CPhonebookEntry(const std::string &_phone, const std::string &_address) :
+		phone(_phone),
+		address(_address) {
+	}
+
+	bool IsMatchingPhone(const std::string &input) {
+		return (input == phone);
+	}
+
+	const std::string &GetAddress() {
+		return address;
+	}
+
+	static bool IsPhoneValid(const std::string &input) {
```

Comment:
(optional)
There's no reason to keep this function as a static method; make it a simple static function in this cpp file.

Reply:
There's was reason to append MODEM_ in front of static function name, but it does not matter - it's a common inconsistency throughout DOSBox code. With code moved to static function it's now cleaner, and future programmers won't have problem e.g. with wrapping this function with a lambda statement, or simply moving the implementation around if it'll ever be necessary.

Thanks! :)